tools/NewRelicHelper.psm1

```diff
@@ -90,40 +90,47 @@ function update_azure_service_configs([System.__ComObject] $project){
 
 		foreach($i in $xml.ServiceConfiguration.ChildNodes){
 			if($i.name -eq $project.Name.ToString()){
-				$modified = $i
+                $modified = $i
 				break
 			}
 		}
-
-		$modifiedConfigSettings = $modified.ConfigurationSettings
-		if ($modifiedConfigSettings -eq $null){
-			# Moved dialog here because if the value already exists, no need to ask again
-			if ($licenseKey -eq $null) {
-				$licenseKey = create_dialog "License Key" "Please enter your New Relic LICENSE KEY"
-			}
-
-			$settingNode.SetAttribute('value', $licenseKey)
-			$modified.AppendChild($configSettingsNode)
-		}
-		else{
-			$nodeExists = $false
-			foreach ($i in $modifiedConfigSettings.Setting){
-				if ($i.name -eq "NewRelic.LicenseKey"){
-					$nodeExists = $true
-				}
-			}
-			if($NewRelicTask -eq $null -and !$nodeExists){
-				# Moved dialog here because if the value already exists, no need to ask again
-				if ($licenseKey -eq $null) {
-					$licenseKey = create_dialog "License Key" "Please enter your New Relic LICENSE KEY"
-				}
-
-				$settingNode.SetAttribute('value', $licenseKey)
-				$modifiedConfigSettings.AppendChild($settingNode)
-			}
-		}
 		
-		$xml.Save($ServiceConfig);
+		#Make sure a matching app was found
+        if($modified -ne $null){
+        
+            $ns = new-object Xml.XmlNamespaceManager $xml.NameTable
+			$ns.AddNamespace('dns', 'http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceConfiguration')
```

Comment:
If there is a potential for this namespace to change over time consider asserting it in the code to short-circuit and give a usable warning or error to quickly identify the issue for consideration.